.github/workflows/build-deploy-pudl.yml

```diff
@@ -33,29 +33,46 @@ jobs:
         uses: actions/checkout@v4
         with:
           ref: ${{ env.BUILD_REF }}
+          fetch-depth: 0
+
+      - name: Skip the build if no changes since the last successful nightly build.
+        run: |
+          CURRENT_COMMIT=$(git rev-parse HEAD)
+          NIGHTLY_COMMIT=$(git rev-parse origin/nightly)
+          if [[ "$CURRENT_COMMIT" == "$NIGHTLY_COMMIT" ]]; then
+            echo "::notice::No changes since last successful nightly build. Skipping."
+            echo "SKIP_BUILD=true" >> $GITHUB_ENV
+            exit 0
+          fi
 
       - name: Set action environment variables
+        if: ${{ env.SKIP_BUILD != 'true' }}
         run: |
           echo "NIGHTLY_TAG=nightly-$(date +%Y-%m-%d)" >> $GITHUB_ENV
           echo "BUILD_ID=$(date +%Y-%m-%d-%H%M)-$(git rev-parse --short HEAD)-${BUILD_REF}" >> $GITHUB_ENV
 
       - name: Show freshly set envvars
+        if: ${{ env.SKIP_BUILD != 'true' }}
         run: |
           echo "GCE_INSTANCE: $GCE_INSTANCE"
           echo "BUILD_REF: $BUILD_REF"
           echo "NIGHTLY_TAG: $NIGHTLY_TAG"
           echo "BUILD_ID: $BUILD_ID"
 
       - name: Tag nightly build
-        if: ${{ (github.event_name == 'schedule') }}
+        if: ${{ (github.event_name == 'schedule') && (env.SKIP_BUILD != 'true') }}
         run: |
           git config user.email "pudl@catalyst.coop"
           git config user.name "pudlbot"
           git tag -a -m "$NIGHTLY_TAG" $NIGHTLY_TAG $BUILD_REF
           git push origin $NIGHTLY_TAG
+          git tag
+          git fetch --tags origin
+          git tag
```

Comment:
Maybe this isn't necessary now that I'm doing `fetch-depth: 0`?

Reply:
I think the most useful debug output would be something like

```
...
git push origin $NIGHTLY_TAG
git fetch --tags origin $NIGHTLY_TAG
git show-ref -d $NIGHTLY_TAG # did the tag make it to origin? 
```

But I sort of feel like all this debug output unnecessary. I expect git push to work, you know? Only when we are questioning the very nature of reality would we need debug output here, and then we can just add it back in.

Reply:
Okay I'll remove it for now and cross my fingers...